app/services/sync_provider_from_find.rb

```diff
@@ -1,53 +1,61 @@
 class SyncProviderFromFind
   def self.call(provider_code:, provider_name: nil, sync_courses: false)
-    new.call(provider_code, provider_name, sync_courses)
+    new(provider_code, provider_name, sync_courses).call
   end
 
-  def call(provider_code, provider_name, sync_courses)
-    provider = create_or_update_provider(provider_name, provider_code, sync_courses)
+  attr_reader :provider_code, :provider_name, :sync_courses
+  attr_accessor :provider
+
+  def initialize(provider_code, provider_name, sync_courses)
+    @provider_code = provider_code
+    @provider_name = provider_name
+    @sync_courses = sync_courses
+  end
+
+  def call
+    @provider = create_or_update_provider
 
     return unless provider.sync_courses?
 
-    # Request provider, all courses and sites.
-    #
-    # For the full response, see:
-    # https://api2.publish-teacher-training-courses.service.gov.uk/api/v3/recruitment_cycles/2020/providers/1N1/?include=sites,courses.sites
-    find_provider = FindAPI::Provider
-      .current_cycle
-      .includes(:sites, courses: [:sites, site_statuses: [:site]])
-      .find(provider_code)
-      .first
+    find_provider = fetch_provider_from_find_api
 
-    update_provider(provider, find_provider)
+    update_provider_details_with_api_response(find_provider)
```

Comment:
I slightly prefer passing method dependencies as arguments vs reading them out of object state... but I think that's a conversation for the pub!